### ISSUE TITLE
Throwing error in serverless function.

### ISSUE THREAD

**Issue by alxmhe** [fauna/faunadb-js#182]
https://github.com/fauna/faunadb-js/blob/6894c0e9926d13cdd8281dd9894814c32e11cfce/src/Client.js#L73
is attempting to require dynamically.
This seems to warn during build of a serverless function (tested with Netlify) :
```
WARNING in /node_modules/faunadb/src/Client.js 73:24-48
Critical dependency: the request of a dependency is an expression
```
and ultimately an error on execution :
```
Response with status 500 in 132 ms.
Error during invocation:  { Error: Cannot find module 'https'
```
Do I need a specific webpack configuration?

**Comment by dijonmusters** [comment]
I am experiencing the same issue when trying to initialise the client in a Netlify function.

```javascript
// functions/test.js
require('dotenv').config();
const faunadb = require('faunadb');
const q = faunadb.query;

const client = new faunadb.Client({
  secret: process.env.FAUNADB_SECRET
});

exports.handler = function(event, context, callback) {
  callback(null, {
    statusCode: 200,
    body: "Hello, World"
  });
}
```
```javascript
// webpack.functions.js
const webpack = require('webpack');

module.exports = {
  plugins: [
    new webpack.DefinePlugin({
      "global.GENTLY": false
    })
  ]
};
```
Running locally with `netlify-lambda serve functions -c ./webpack.functions.js`
```
// Error running locally
Error during invocation:  Error: Cannot find module 'https'
    at a (/Users/jon/repos/test-fauna-netlify/lambda/test.js:31:192106)
    at new m (/Users/jon/repos/test-fauna-netlify/lambda/test.js:19:7867)
    at Object.<anonymous> (/Users/jon/repos/test-fauna-netlify/lambda/test.js:19:6093)
    at a (/Users/jon/repos/test-fauna-netlify/lambda/test.js:1:158)
    at /Users/jon/repos/test-fauna-netlify/lambda/test.js:1:957
    at Object.<anonymous> (/Users/jon/repos/test-fauna-netlify/lambda/test.js:1:967)
    at Module._compile (internal/modules/cjs/loader.js:936:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:947:10)
    at Module.load (internal/modules/cjs/loader.js:790:32)
    at Function.Module._load (internal/modules/cjs/loader.js:703:12)
    at Module.require (internal/modules/cjs/loader.js:830:19)
    at require (internal/modules/cjs/helpers.js:68:18)
    at /Users/jon/repos/test-fauna-netlify/node_modules/netlify-lambda/lib/serve.js:120:17
    at Layer.handle [as handle_request] (/Users/jon/repos/test-fauna-netlify/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/jon/repos/test-fauna-netlify/node_modules/express/lib/router/route.js:137:13)
    at next (/Users/jon/repos/test-fauna-netlify/node_modules/express/lib/router/route.js:131:14) {
  code: 'MODULE_NOT_FOUND'
}
```
```
// Error running deployed version on Netlify
{
  "errorMessage": "Cannot find module 'https'",
  "errorType": "Error",
  "stackTrace": [
    "new m (/var/task/test.js:19:7867)",
    "Object.<anonymous> (/var/task/test.js:19:6093)",
    "a (/var/task/test.js:1:220)",
    "/var/task/test.js:1:1019",
    "Object.<anonymous> (/var/task/test.js:1:1029)",
    "Module._compile (module.js:652:30)",
    "Object.Module._extensions..js (module.js:663:10)",
    "Module.load (module.js:565:32)",
    "tryModuleLoad (module.js:505:12)"
  ]
}
```

**Comment by WvanDam** [comment]
Experiencing the same, with the same setup as @dijonmusters.

**Comment by WvanDam** [comment]
This line was introduced in version 2.81. However, rolling back to version 2.8.0 did not fix the second issue, regarding the https module.

**Comment by WvanDam** [comment]
2.7.0 appears to do the trick

**Comment by danieltodonnell** [comment]
We're focussing on adding new drivers functionality and addressing the drivers issue people have been experiencing next week.  If using the 2.7 fallback is difficult, it hopefully will only be for a short time.

Thank you for your patience.

**Comment by dijonmusters** [comment]
Thanks for looking into this so quickly @danieltodonnell 

I can confirm that the rollback to 2.7.0 is working for me 🙂

**Comment by aequasi** [comment]
Can also replicate this when using webpack targeting `webworker` (cloudflare workers)

**Comment by eigilsagafos** [comment]
@danieltodonnell Any news on this?

**Comment by danieltodonnell** [comment]
Yes!  This has had all our minds on it for the past few days, and we have from an optional keep alive flag to a much more involved solution that we've been testing thoroughly.  The good news is: we have developed a solution that we believe will solve this entirely.

It will be released in our `v2.9` JS driver next week

**Comment by eigilsagafos** [comment]
Great! Thanks for the update @danieltodonnell

**Comment by danieltodonnell** [comment]
For anyone blocked by this issue, I am pleased to report that you should now be UN-blocked as-of version faunadb@2.9.2

https://www.npmjs.com/package/faunadb/v/2.9.2

This has been rigorously tested to have a smooth experience. If you experience any issues, feel free to tell us in Fauna Slack Community channel.

We apologize for the delay and inconvenience. Thank you for your patience.

**Comment by danieltodonnell** [comment]
I'm closing this issue but let us know if there are further issues

**Comment by yet-another-dev** [comment]
I'm seeing the following error while trying to use the driver in a [Cloudflare Worker](https://workers.cloudflare.com/): 

```
ReferenceError: XMLHttpRequest is not defined
```

@aequasi [Sounds like](https://github.com/fauna/faunadb-js/issues/182#issuecomment-533427561) you're using Workers, are you seeing this issue?

I can open a new issue for this if that would be best.
